Svc/BufferAccumulator/ArrayFIFOBuffer.cpp

```diff
@@ -10,98 +10,93 @@
 
+void BufferAccumulator::ArrayFIFOBuffer ::init(Fw::Buffer* const elements,
+                                               NATIVE_UINT_TYPE capacity) {
+  this->elements = elements;
+  this->capacity = capacity;
+
+  // Construct all elements
+  for (NATIVE_UINT_TYPE idx = 0; idx < capacity; idx++) {
+      new (&this->elements[idx]) Fw::Buffer();
   }
+}
 
-  // ----------------------------------------------------------------------
-  // Public functions
-  // ----------------------------------------------------------------------
+bool BufferAccumulator::ArrayFIFOBuffer ::enqueue(const Fw::Buffer& e) {
 
-  void BufferAccumulator::ArrayFIFOBuffer ::
-    init(Fw::Buffer *const elements, NATIVE_UINT_TYPE capacity)
-  {
-    this->elements = elements;
-    this->capacity = capacity;
+  if (this->elements == nullptr) {
+    return false;
   }
 
-  bool BufferAccumulator::ArrayFIFOBuffer ::
-    enqueue(const Fw::Buffer& e)
-  {
-    if (this->elements == nullptr) {
-      return false;
-    }
-    bool status;
-    if (this->size < this->capacity) {
-      // NOTE(mereweth) enqueueIndex is unsigned, no need to compare with 0
-      FW_ASSERT(enqueueIndex < this->capacity, enqueueIndex);
-      this->elements[this->enqueueIndex] = e;
-      this->enqueueIndex = (this->enqueueIndex + 1) % this->capacity;
-      status = true;
-      ++this->size;
-    }
-    else {
-      status = false;
-    }
-    return status;
+  bool status;
+  if (this->size < this->capacity) {
+    // enqueueIndex is unsigned, no need to compare with 0
+    FW_ASSERT(enqueueIndex < this->capacity, enqueueIndex);
+    this->elements[this->enqueueIndex] = e;
+    this->enqueueIndex = (this->enqueueIndex + 1) % this->capacity;
+    status = true;
+    this->size++;
+  } else {
+    status = false;
   }
 
-  bool BufferAccumulator::ArrayFIFOBuffer ::
-    dequeue(Fw::Buffer& e)
-  {
-    if (this->elements == nullptr) {
-      return false;
-    }
-    FW_ASSERT(this->elements);
-    bool status;
-    if (this->size > 0) {
-      // NOTE(mereweth) dequeueIndex is unsigned, no need to compare with 0
-      FW_ASSERT(dequeueIndex < this->capacity, enqueueIndex);
-      e = this->elements[this->dequeueIndex];
-      this->dequeueIndex = (this->dequeueIndex + 1) % this->capacity;
-      --this->size;
-      status = true;
-    }
-    else {
-      status = false;
-    }
-    return status;
-  }
+  return status;
+}
 
-  U32 BufferAccumulator::ArrayFIFOBuffer ::
-    getSize() const
-  {
-    return this->size;
+bool BufferAccumulator::ArrayFIFOBuffer ::dequeue(Fw::Buffer& e) {
+
+  if (this->elements == nullptr) {
+    return false;
   }
 
-  U32 BufferAccumulator::ArrayFIFOBuffer ::
-    getCapacity() const
-  {
-    return this->capacity;
+  FW_ASSERT(this->elements);
+  bool status;
+
+  if (this->size > 0) {
+    // dequeueIndex is unsigned, no need to compare with 0
+    FW_ASSERT(dequeueIndex < this->capacity, enqueueIndex);
```

Comment:
Would it be more appropriate to pass `dequeueIndex` instead of `enqueueIndex` as the second argument to `FW_ASSERT ?